src/harmaja.ts

```diff
@@ -624,6 +624,38 @@ function replaceAll(controller: NodeController | null, oldContent: HarmajaStatic
     //console.log("Replaced " + debug(oldContent) + " with " + debug(newContent))
 }
 
+function replaceAllInPlace(controller: NodeController, oldNodes: ChildNode[], newNodes: ChildNode[], createdNodes: ChildNode[], deletedNodes: ChildNode[]) {
+    if (oldNodes.length === 0) throw new Error("Cannot replace zero nodes")
+    if (newNodes.length === 0) throw new Error("Cannot replace with zero nodes")
+
+    const oldFirst = controller.currentElements[0]
+    controller.currentElements = newNodes
+    detachController(deletedNodes, controller)
+    attachController(createdNodes, controller, undefined, true)
+
+    if (controller.unsub) {
+        // Make sure newNodes[0] still calls controller.unsub on unmount
+        if (deletedNodes.includes(oldFirst)) {
+            attachOnUnmount(newNodes[0], controller.unsub)
+        } else if (newNodes[0] !== oldNodes[0]) {
+            detachOnUnmount(oldNodes[0], controller.unsub)
+            attachOnUnmount(newNodes[0], controller.unsub)
+        }
+    }
+
+    // oldNodes[0] may be the controller's initial text node, which hasn't been
+    // added to DOM before the first render. Thus `parentElement?`.
+    oldNodes[0].parentElement?.replaceChild(newNodes[0], oldNodes[0])
+    deletedNodes.forEach(node => node.remove())
+    for (let i = 1; i < newNodes.length; i++) {
+        newNodes[i - 1].after(newNodes[i])
+    }
```

Comment:
Here, the number of DOM operations is actually not minimal. I guess a diffing algorithm could be employed to minimize the number of `node.after()` calls, but don't know if that has any effect on performance really.